feat(ui): add bookmark generation option inside viewer

milesap opened the pull request:
Share button appears in left-side menu, and can be enabled with a
'shareable' config option. Can also obtain short urls from Google API if
API key is set.

Closes #1181

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1226)
<!-- Reviewable:end -->

alyec commented:
Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSal9g5TdRS0yzOcayv:b1nxcg6) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
> ```JSON
> 
>     "properties": {
>         "version": { "type": "string", "enum": [ "1.2.0" ], "description": "The schema version used to validate the configuration file.  The schema should enumerate the list of versions accepted by this version of the viewer." },
> ```

If this is 1.2.0 it should probably have compatibility with 1.1 and 1.0 schemas; if it does not it should be a 2.0 schema.

---

*[schemas/fgpv_schema_v1_2_0.json, line 671 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSalHcrJQ7FHMwY0BeQ:-KSalHcrJQ7FHMwY0BeR:b-viyayr) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L671)):*
> ```JSON
>         "theme": { "type": "string", "enum": [ "default" ], "default": "default", "description": "UI theme of the viewer" },
>         "logoUrl": { "type": "string", "description": "An optional image to be used in the place of the default viewer logo" },
>         "shareable": { "type": "boolean", "default": true, "description": "Allow users to generate bookmark urls." },
> ```

I think we should make this opt-in instead of opt-out since this breaks in the embedded use case.  The description should also be "Allow user to generate bookmark URLs from within the map application" since I'm assuming this doesn't disable the bookmark API calls if it's false.

---

*[src/schema.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapt_owYWKc8Xv12q_:-KSapt_owYWKc8Xv12qa:b-lzevu) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/schema.json#L667)):*
> ```JSON
> 
>     "properties": {
>         "version": { "type": "string", "enum": [ "1.2.0" ], "description": "The schema version used to validate the configuration file.  The schema should enumerate the list of versions accepted by this version of the viewer." },
> ```

same comments as above

---

*[src/app/ui/sidenav/sidenav.service.js, line 1 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSam4eZIDt4UfmDtuik:-KSam4eZIDt4UfmDtuil:b-1yni8y) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L1)):*
> ```JavaScript
> /* global RV */
> ```

I think we have a service for accessing the global without explicitly declaring it.

---

*[src/app/ui/sidenav/sidenav.service.js, line 44 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSamUg3PRSy6lSfJHYM:-KSamUg3PRSy6lSfJHYN:b-fz48e) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L44)):*
> ```JavaScript
>             // url cache to avoid unneeded API calls
>             const URLS = {
>                 short: undefined,
> ```

If I change the map

---

*[src/app/ui/sidenav/sidenav.service.js, line 93 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSanUch3ZBQOcIGxx0o:-KSanUch3ZBQOcIGxx0p:b5p2t9r) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L93)):*
> ```JavaScript
>                     $http.post(self.googleAPIUrl, { longUrl: self.url }).then(
>                         r => URLS.short = self.url = r.data.id,
>                         () => URLS.short = undefined // reset cache from failed API call
> ```

Shouldn't this be a separate `.catch` (if the exception happens within the resolve part I believe everything falls through)?

---

*[src/app/ui/sidenav/sidenav.service.js, line 110 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapR_M5-zPYq6SYcEZ:-KSapR_M5-zPYq6SYcE_:botm8bi) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L110)):*
> ```JavaScript
>                 $(event.currentTarget).select();
>                 // try to automatically copy link - if successful display message
>                 self.linkCopied = document.execCommand('copy');
> ```

From the demo: I think the select is fine, but I don't think it should overwrite the clipboard by clicking on a text box.  Although I do realize that most people will want to copy the bookmark as soon as they create it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 10 of 15 files at r1.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[src/app/layout/shell.directive.js, line 115 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSarQgd_4IIJYrNRUDA:-KSarQgd_4IIJYrNRUDB:bb7pcsz) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/layout/shell.directive.js#L115)):*
> ```JavaScript
>                             disableParentScroll: false,
>                             targetEvent: event,
>                             clickOutsideToClose: false,
> ```

true

---

*[src/app/ui/sidenav/share-dialog.html, line 1 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSavKgjNHjyxkeKa_io:-KSavKgjNHjyxkeKa_ip:b7e9z0f) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/share-dialog.html#L1)):*
> ```HTML
> <md-dialog aria-label="{{'sidenav.label.sharelink' | translate}}">
> ```

Use the same styles for the dialog as are used for the Help dialog (white header, square corners, etc.) and truncate the bookmark link with an ellipsis.

---

*[src/app/ui/sidenav/sidenav.service.js, line 110 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapR_M5-zPYq6SYcEZ:-KSatUfS4zQnomyR11t7:b1tsm24) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L110)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> From the demo: I think the select is fine, but I don't think it should overwrite the clipboard by clicking on a text box.  Although I do realize that most people will want to copy the bookmark as soon as they create it.
</details>
What about adding an explicit button to copy the bookmark or a notification telling the user what will happen once clicked:

![](https://i.imgur.com/zZZtg6S.png) 

or

![](https://i.imgur.com/8oJvjH3.png)

---

*[src/content/styles/modules/_side-nav.scss, line 2 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSasBw5_knMdLoKs3nn:-KSasBw5_knMdLoKs3no:bxzg43k) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/content/styles/modules/_side-nav.scss#L2)):*
> ```SCSS
> @mixin side-nav {
>     md-dialog {
> ```

This will affect all dialogs in the viewer.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

alyec commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[src/app/ui/sidenav/sidenav.service.js, line 110 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapR_M5-zPYq6SYcEZ:-KSaw42955VSGoJzT635:bqjvelx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L110)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> What about adding an explicit button to copy the bookmark or a notification telling the user what will happen once clicked:
> 
> ![](https://i.imgur.com/zZZtg6S.png) 
> 
> or
> 
> ![](https://i.imgur.com/8oJvjH3.png)
</details>
:white_check_mark: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSb0FUj3ODX2JV4G_gV:bbelz6f) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> If this is 1.2.0 it should probably have compatibility with 1.1 and 1.0 schemas; if it does not it should be a 2.0 schema.
</details>
I would say it is compatible with 1.1 and 1.0 since it's adding an optional config parameter. Unless making it opt-out like you mentioned below would make this a breaking change? Please clarify

---

*[schemas/fgpv_schema_v1_2_0.json, line 671 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSalHcrJQ7FHMwY0BeQ:-KSb19KdJ_F3u5Xf3pO-:b8b3gsn) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L671)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> I think we should make this opt-in instead of opt-out since this breaks in the embedded use case.  The description should also be "Allow user to generate bookmark URLs from within the map application" since I'm assuming this doesn't disable the bookmark API calls if it's false.
</details>
Ok so something more along the lines of `disableShareable` - if not set it is enabled by default?

---

*[src/app/ui/sidenav/sidenav.service.js, line 110 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapR_M5-zPYq6SYcEZ:-KSb1mNvO-3MfNgcbB2D:bweylvv) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L110)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> :white_check_mark: 
</details>
Looks like we're going with the second option "Click to copy". Should we still select the text on focus for keyboard users?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

alyec commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSb2NsfIba7uRIwHQap:b-azhn15) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> I would say it is compatible with 1.1 and 1.0 since it's adding an optional config parameter. Unless making it opt-out like you mentioned below would make this a breaking change? Please clarify
</details>
It should be, but I've been out of the loop for a bit so I wanted to clarify in case there were any other changes I missed.

---

*[schemas/fgpv_schema_v1_2_0.json, line 671 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSalHcrJQ7FHMwY0BeQ:-KSb2VdFyWni-UUCUnUZ:b-1phx3z) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L671)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Ok so something more along the lines of `disableShareable` - if not set it is enabled by default?
</details>
No, I'm thinking just change the default to false.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[src/app/ui/sidenav/sidenav.service.js, line 110 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapR_M5-zPYq6SYcEZ:-KSb39QpRtC9RAb8sIIU:b-fognpb) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L110)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Looks like we're going with the second option "Click to copy". Should we still select the text on focus for keyboard users?
</details>
No, if you tab to the field, nothing should happen.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 8 of 15 files reviewed at latest revision, 10 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSbGoFrPHJUPqrOja8F:b-ybf5rd) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> It should be, but I've been out of the loop for a bit so I wanted to clarify in case there were any other changes I missed.
</details>
Keeping with 1.2.0 as this change is compatible with 1.1 and 1.0 schemas (it would be omitted and no share url option is given).

---

*[schemas/fgpv_schema_v1_2_0.json, line 671 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSalHcrJQ7FHMwY0BeQ:-KSbGxl4rAxfukcINPWJ:b6d8xu) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L671)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> No, I'm thinking just change the default to false.
</details>
Ok changed to `false` and updated the description

---

*[src/schema.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapt_owYWKc8Xv12q_:-KSbIPnmqfKqz8KivNSE:b-62bz1c) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/schema.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> same comments as above
</details>
Made same changes here as above. 

---

*[src/app/layout/shell.directive.js, line 115 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSarQgd_4IIJYrNRUDA:-KSbIwlfZ2IdSk3CZSKl:b499q7j) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/layout/shell.directive.js#L115)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> true
</details>
Changed

---

*[src/app/ui/sidenav/share-dialog.html, line 1 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSavKgjNHjyxkeKa_io:-KSbSrhu-tnUrV--lqqg:b6cjxgc) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/share-dialog.html#L1)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Use the same styles for the dialog as are used for the Help dialog (white header, square corners, etc.) and truncate the bookmark link with an ellipsis.
</details>
Ok changed the style of the dialog and added the ellipsis.

---

*[src/app/ui/sidenav/sidenav.service.js, line 1 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSam4eZIDt4UfmDtuik:-KSbVTU-0XuLv7F6m7J6:biphg68) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L1)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> I think we have a service for accessing the global without explicitly declaring it.
</details>
Good to know, replaced `RV` with `globalRegistry`

---

*[src/app/ui/sidenav/sidenav.service.js, line 44 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSamUg3PRSy6lSfJHYM:-KSbSRtJ7TIOSIED2aS-:bted7qs) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L44)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> If I change the map
</details>
I'm assuming the map won't change while the share dialog is opened - if this is the case there is no issue since moving the map then re-opening will give the correct link.

---

*[src/app/ui/sidenav/sidenav.service.js, line 93 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSanUch3ZBQOcIGxx0o:-KSbS5at3fGVe3xKkv4F:b-dnfj6v) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L93)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> Shouldn't this be a separate `.catch` (if the exception happens within the resolve part I believe everything falls through)?
</details>
Changed to `catch` for improved readability - the second argument of `.then` can also be used for the errorCallback

---

*[src/app/ui/sidenav/sidenav.service.js, line 110 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSapR_M5-zPYq6SYcEZ:-KSbRJrvuVIRgZOQYXTU:b-ps1x07) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L110)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> No, if you tab to the field, nothing should happen.
</details>
Added a click to copy hint

---

*[src/content/styles/modules/_side-nav.scss, line 2 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSasBw5_knMdLoKs3nn:-KSbRPWQMP4DxY3GzWPi:b-ytt50o) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/content/styles/modules/_side-nav.scss#L2)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> This will affect all dialogs in the viewer.
</details>
Changed to its class name instead of `md-dialog`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSbWZma_rkP8cSEjegB:b-3ak2iv) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Keeping with 1.2.0 as this change is compatible with 1.1 and 1.0 schemas (it would be omitted and no share url option is given).
</details>
So this should become `["1.2","1.1","1.0"]` (this is the list of schema versions which this version of the viewer can accept), I believe there is a separate version property somewhere which specifies the version of the schema itself.  The idea would be that if I try to build FGPV with a 1.1 sample page it should work just fine and will validate.

---

*[src/app/ui/sidenav/sidenav.service.js, line 44 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSamUg3PRSy6lSfJHYM:-KSbXv1gYv6CThvbxfcV:b7o6xhn) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L44)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> I'm assuming the map won't change while the share dialog is opened - if this is the case there is no issue since moving the map then re-opening will give the correct link.
</details>
Yeah, sorry for the partial comment and thanks for figuring out the missing bit.

---

*[src/app/ui/sidenav/sidenav.service.js, line 93 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSanUch3ZBQOcIGxx0o:-KSbXa87rsJBGQnPcqxv:b-gotypy) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L93)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Changed to `catch` for improved readability - the second argument of `.then` can also be used for the errorCallback
</details>
Sorry, I'd like to add a style suggestion:
```js
  $http.post(self.googleAPIUrl, { longUrl: self.url })
    .then(r => URLS.short = self.url = r.data.id)
    .catch(() => URLS.short = undefined);
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 3 of 7 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/content/styles/modules/_side-nav.scss, line 2 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSasBw5_knMdLoKs3nn:-KSbZl1p1w7HoZXC3Cm-:bnde1tg) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/content/styles/modules/_side-nav.scss#L2)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Changed to its class name instead of `md-dialog`
</details>
Just a tiny change: 30% and 40% seem a bit too much. Can you set it to  200 by 400 px for now? We will sort the size out when defining mobile layouts.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 11 of 15 files reviewed at latest revision, 5 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSbap9FdB9g4BvLmoil:b-nl7bz7) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> So this should become `["1.2","1.1","1.0"]` (this is the list of schema versions which this version of the viewer can accept), I believe there is a separate version property somewhere which specifies the version of the schema itself.  The idea would be that if I try to build FGPV with a 1.1 sample page it should work just fine and will validate.
</details>
Ah makes sense now - updated

---

*[src/app/ui/sidenav/sidenav.service.js, line 44 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSamUg3PRSy6lSfJHYM:-KSbazhOa_8JKNLQ0Vnp:b-8fx2u7) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L44)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> Yeah, sorry for the partial comment and thanks for figuring out the missing bit.
</details>
:+1: 

---

*[src/app/ui/sidenav/sidenav.service.js, line 93 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSanUch3ZBQOcIGxx0o:-KSbbWMXc9Nqz1AZdvcp:b-v29y6i) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/app/ui/sidenav/sidenav.service.js#L93)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> Sorry, I'd like to add a style suggestion:
> ```js
>   $http.post(self.googleAPIUrl, { longUrl: self.url })
>     .then(r => URLS.short = self.url = r.data.id)
>     .catch(() => URLS.short = undefined);
> ```
</details>
Looks good - updated

---

*[src/content/styles/modules/_side-nav.scss, line 2 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSasBw5_knMdLoKs3nn:-KSbcDBf4mNevbtbiaHP:br5j78v) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/src/content/styles/modules/_side-nav.scss#L2)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Just a tiny change: 30% and 40% seem a bit too much. Can you set it to  200 by 400 px for now? We will sort the size out when defining mobile layouts.
</details>
np - updated

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 1 of 15 files at r1, 1 of 7 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSbd1dL51hKvPY7W-jB:b-skkzxx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Ah makes sense now - updated
</details>
Should `fgpv_schema_v1_1_0.json` file be updated the same way as well? 
`"enum": [ "1.1.0","1.0.0" ]`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

alyec commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSbdHwVqvjH-QnaM6yK:b80gjti) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Should `fgpv_schema_v1_1_0.json` file be updated the same way as well? 
> `"enum": [ "1.1.0","1.0.0" ]`
</details>
It should have been.  If it's not deployed yet then I'd be okay with fixing it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSbeVAyzfAhc1kghBi6:blb8myv) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> It should have been.  If it's not deployed yet then I'd be okay with fixing it.
</details>
BTW schema versioning doesn't make sense with a patch version, if you change the schema you've pretty much made a minor version update regardless of how small it is (so `1.2` for schemas `1.2.0` for the application).  The version inside the schema can be `1.2.1` if you're fixing a typo in the docs or something like that, but I don't think that needs to be added to the compatibility list.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

jolevesq commented:
Quick question, when we share link should it remember the extent and the list of layers? If I remove a layer should it remove it inside the bookmark?

---

Reviewed 8 of 15 files at r1, 3 of 7 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-:-KSfSXuXTEjldPIuxZlM:b-qf7sr8)*
<!-- Sent from Reviewable.io -->

milesap commented:
Removing layers does not get saved in the bookmark. You can set visibility however which is saved.

---

Review status: 6 of 15 files reviewed at latest revision, 2 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSfwAFDgo1jH6Aefyoa:b-vzs4t4) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> BTW schema versioning doesn't make sense with a patch version, if you change the schema you've pretty much made a minor version update regardless of how small it is (so `1.2` for schemas `1.2.0` for the application).  The version inside the schema can be `1.2.1` if you're fixing a typo in the docs or something like that, but I don't think that needs to be added to the compatibility list.
</details>
Removed patch from schema versions. Since `fgpv_schema_v1_1_0.json` is deployed did not update.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-:-KSfwWDvS4hQM9Y6V8ra:bg0zv3c)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 9 of 9 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSg01OSEkmWDEuY6xTB:bntbjyn) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary>
> Removed patch from schema versions. Since `fgpv_schema_v1_1_0.json` is deployed did not update.
</details>
Schema 1.0.0 uses the two point numbering but schema 1.1.0 uses three point so this should be `[ "1.2","1.1.0","1.0" ]`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 13 of 15 files reviewed at latest revision, 2 unresolved discussions.

---

*[schemas/fgpv_schema_v1_2_0.json, line 667 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-KSal9g5TdRS0yzOcayu:-KSg1HoI0e-TAAIEBdRR:b32w1t) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/ed2d376b3fbb6c4aee1d510506b7201cedc13085/schemas/fgpv_schema_v1_2_0.json#L667)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary>
> Schema 1.0.0 uses the two point numbering but schema 1.1.0 uses three point so this should be `[ "1.2","1.1.0","1.0" ]`.
</details>
Ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

jolevesq commented:
When I try from your branch, visibility is not saved (I look at index-one). Should I look at another one? It would be very to bookmark the extent as well. Should this be add as a new issue?

---

Reviewed 7 of 9 files at r4, 2 of 2 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-:-KShHsXyTED30yhpbalR:b-cqjkhk)*
<!-- Sent from Reviewable.io -->

milesap commented:
You can [test bookmarking](http://fgpv.cloudapp.net/demo/users/milesap/1181-generate-display-bookmark/samples/index-fgp-en.html?keys=Airports) with `index-fgp-en.html` or `index-fgp-fr.html`. The host page requires some additional javascript to be included for bookmarking to be restored (`index-one.html` does not have this).

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-:-KSkPAP1OnAjkvwALAqO:brwr03m)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 7 of 9 files at r4, 2 of 2 files at r5, 3 of 3 files at r6.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

milesap commented:
Bumped schema to `v1.3` and included missing `v1.2` file along with `v1.3` in schema folder

---

Review status: 6 of 16 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226#-:-KTz0rzXRAvSzltWfN-0:b-cy11dr)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 11 of 11 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 11 of 11 files at r7.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1226)*
<!-- Sent from Reviewable.io -->
